api/src/pcapi/core/offers/models.py

```diff
@@ -372,7 +379,7 @@ class Offer(PcObject, Model, ExtraDataMixin, DeactivableMixin, ValidationMixin,
 
     authorId = sa.Column(sa.BigInteger, sa.ForeignKey("user.id"), nullable=True)
 
-    author = sa.orm.relationship("User", foreign_keys=[authorId], backref="offers")  # type: ignore [misc]
+    author: Mapped["User"] = relationship("User", foreign_keys=[authorId], backref="offers", uselist=False)
```

Comment:
... par contre ici le champ est nullable, donc on devrait plutôt avoir quelque chose ça, je pense : 

```suggestion
    author: Mapped[typing.Optional["User"]] = relationship("User", foreign_keys=[authorId], backref="offers", uselist=False)
```